### ISSUE TITLE
Implement Google authentication and scope authorization on Breadboard Web

### ISSUE THREAD

**Issue by dglazkov** [breadboard-ai/breadboard#938]
We really need to get Google authentication and scope authorization implemented for Breadboard web app. This would allow us build graphs that talk to Google APIs and use credentials and tokens properly.

The current [credentials](https://github.com/breadboard-ai/breadboard/blob/8dc4e004f6f6198d083c3f7c9fc579791ffc8ecc/packages/node-nursery-web/src/nodes/credentials.ts) node is a dead end in this respect: it presumes that it runs in the browser and that is too limiting. Instead, we likely need an approach where we teach the [secrets](https://github.com/breadboard-ai/breadboard/blob/8dc4e004f6f6198d083c3f7c9fc579791ffc8ecc/packages/core-kit/src/nodes/secrets.ts#L96) node to deal with complex secrets (like, `GOOGLE_AUTH` key + a list of scopes) and let the app take care of the authentication.

**Comment by Mearman** [comment]
- this could be useful for the Drive access version of this ticket https://github.com/breadboard-ai/breadboard/issues/927

**Comment by zeroasterisk** [comment]
Agreed, was dusting off my use cases, and I cannot really proceed without a bearer token scoped to my application or user credentials :(

I'm spoiled with colab and...

```python
import sys

if "google.colab" in sys.modules:
    from google.colab import auth as google_auth

    google_auth.authenticate_user()

from google.auth import default

creds, _ = default()
```

Could we do something which is (to the user) as simple?

**Comment by Mearman** [comment]
@zeroasterisk I haven't finishing building it yet. But I have ended up using a Service Account with domain wide delegation to generate an OAuth token headlessly

**Comment by dglazkov** [comment]
We have a path with the connection server, need to get there.

cc @aomarks
